lib/service/storage/elasticsearch.js

```diff
@@ -3154,14 +3156,36 @@ class ElasticSearch extends Service {
           },
         },
         index: await this._getAvailableIndice(index, HIDDEN_COLLECTION),
-      });
+      };
+
+      esRequest.wait_for_active_shards = await this._getWaitForActiveShards();
```

Comment:
This should be added directly at `esRequest` initialisation to avoid unecessary hidden class